src/ORM/DataObject.php

```diff
@@ -3585,6 +3590,7 @@ public function flushCache(bool $persistent = true): static
             if (isset(DataObject::$_cache_get_one[$class])) {
                 unset(DataObject::$_cache_get_one[$class]);
             }
+            DataList::reset($class);
```

Comment:
The changes here, `flush_and_destroy_cache()`, and `reset()` ensure that the new cache will be invalidated under the same conditions that the old `get_one()` cache gets invalidated.